src/nih_wayfinding/app/scripts/views/navigate/navigate-controller.js

```diff
@@ -24,11 +27,15 @@
         initialize();
 
         function initialize() {
+            setDefaultFooter();
             NavbarConfig.set({
                 title: 'Navigate Route',
                 back: 'routing'
             });
             ctl.map = Map;
+            ctl.origin = Config.stubs.geolocation;
```

Comment:
Is this branch rebased against current develop? There should be a Navigation.setFirst() call here and then the current position comes back as the argument to the 'nih.navigation'positionUpdated' event, rather than saving origin to scope.

Remember, you can always attach multiple event handlers to a single $scope.$on event if you want, to keep the logic separated